cmd/juju/model/addgeneration.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package model
+
+import (
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+
+	"github.com/juju/juju/api/modelgeneration"
+	jujucmd "github.com/juju/juju/cmd"
+	"github.com/juju/juju/cmd/modelcmd"
+)
+
+const (
+	addGenerationSummary = "Adds a new generation to the model."
+	addGenerationDoc     = `
```

Comment:
There is a card to refine the help text:  https://trello.com/c/QLFkHlN9/922-refine-text-for-add-generation-cmd-help